invenio_theme/templates/invenio_theme/header_login.html

```diff
@@ -24,7 +24,7 @@
 {%- if config['ACCOUNTS'] %}
 {%- if not current_user.is_authenticated %}
 <form class="navbar-form navbar-right">
-  <a href=" {{url_for_security('login')}}" class="btn btn-default"><i class="fa fa-sign-in"></i> {{_('Log in')}}</a>
+  <a href=" {{url_for_security('login', next=request.path)}}" class="btn btn-default"><i class="fa fa-sign-in"></i> {{_('Log in')}}</a>
```

Comment:
Actually the login button is never displayed on the login page, so it shouldn't be necessary.